### PR TITLE
Port #5755 to feature/atree-inlining-cadence-v0.42

### PR DESCRIPTION
**Pull request by turbolent**
Work towards https://github.com/onflow/cadence/issues/3220

Port #5755 to https://github.com/onflow/flow-go/tree/feature/atree-inlining-cadence-v0.42

Work done by @fxamacker

This PR adds `--fix-testnet-slabs-with-broken-references` flag to the migration program.

It uses a feature from onflow/atree#388 to fix 10 testnet registers affecting 9 testnet accounts.

The 9 testnet accounts are hardcoded in this commit to prevent accidentally applying this fix to any other accounts.

In testnet, broken references seem to have resulted from a bug that was fixed 2 years ago by onflow/cadence#1565.
    
A broken reference is a `StorageID` referencing a non-existent register.  So far, only 10 registers in testnet (none on mainnet) were found to contain broken references.

### CODE REVIEW

**Comment by turbolent** [comment]
@fxamacker Integrated the migration in f9798b7278d62318ce5f1a7d023e7fd9c18059d5

File: cmd/util/cmd/execution-state-extract/execution_state_extract.go [hunk 1]
```diff
@@ -351,32 +365,52 @@ func newMigrations(
...
+				),
+			)
+		}
+
+		accountBasedMigrations = append(
+			accountBasedMigrations,
+			migrators.NewFixBrokenReferencesInSlabsMigration(
+				rwf,
+				migrators.TestnetAccountsWithBrokenSlabReferences,
+			),
```
Comment (fxamacker) [inline cmd/util/cmd/execution-state-extract/execution_state_extract.go:391]: `FixBrokenReferencesInSlabsMigration` is added already under a flag, so we don't need it here.

```suggestion
```

**Review by fxamacker — COMMENTED** [review]
Nice!  Thanks for porting this over!

I left a comment about integration with pipeline.

File: cmd/util/cmd/execution-state-extract/execution_state_extract.go (hunk 1 above)
Comment (turbolent) [reply cmd/util/cmd/execution-state-extract/execution_state_extract.go:391]: Ah, good catch!

**Review by fxamacker — APPROVED** [review]

**Comment by fxamacker** [comment]
@turbolent There are some compile errors due to breaking compatibility of atree.  

Do you mind if I push some commits to fix them?

File: cmd/util/ledger/migrations/fix_broken_data_migration.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+) (
+	newPayloads []*ledger.Payload,
+	err error,
+) {
+
+	if _, exist := m.accountsToFix[address]; !exist {
+		return oldPayloads, nil
+	}
+
+	migrationRuntime, err := NewAtreeRegisterMigratorRuntime(address, oldPayloads)
```
Comment (SupunS) [inline cmd/util/ledger/migrations/fix_broken_data_migration.go:67]: I recall some previous conversation where we said creating a migration runtime with all payloads can be very expensive (not sure how expensive that is though). Can the runtime/storage here be created using only the payloads belongs to the accounts in `m.accountsToFix`, as an optimization?

**Review by SupunS — APPROVED** [review]

**Comment by turbolent** [comment]
[6f7004c](https://github.com/onflow/flow-go/pull/5765/commits/6f7004c6746fea37498d7d25a044072cde0f6500) and [e5c6ec2](https://github.com/onflow/flow-go/pull/5765/commits/e5c6ec23a155f17b3c943cec975b68e66407bb7a) look great @fxamacker, thank you for fixing this up!

**Review by fxamacker — APPROVED** [review]

**Pull request merged**